File: Demo_Java/src/main/java/com/zuo/demo_java/MainFragment.java
package com.zuo.demo_java;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import com.zuo.demo_java.databinding.FragmentMainBinding;


/**
 * @author zuo
 * @date 2020/7/26 22:28
 */
public class MainFragment extends Fragment {

    private FragmentMainBinding binding;
    private NavController navController;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_main, container, false);
        binding = DataBindingUtil.bind(view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        navController = NavHostFragment.findNavController(this);
        initView();
    }

    private void initView() {
        binding.msgView.setOnClickListener(view -> {
            navController.navigate(R.id.action_mainFragment_to_msgFragment);
        });
    }
}

File: app/src/main/kotlin/com/zuo/kotlindemo/main/MainFragment.kt
package com.zuo.kotlindemo.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment.findNavController
import androidx.navigation.get
import com.zuo.kotlindemo.R
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.fragment_main.*

class MainFragment : Fragment(), View.OnClickListener {
    //获取导航控制器
    private val navController: NavController
        get() {
            return findNavController(nav_host_fragment)
        }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_main, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        login_view.setOnClickListener(this)
        uav_view.setOnClickListener(this)
        msg_view.setOnClickListener(this)
    }

    override fun onClick(p0: View?) {
        when (p0) {
            uav_view -> {
                //跳转到无人机成果列表
                navController.navigate(R.id.action_mainFragment_to_uavResultFragment)
            }
            msg_view -> {
                //跳转到聊天界面
                navController.navigate(R.id.action_mainFragment_to_msgListFragment)
            }
            login_view -> {
                //跳转登录界面
                navController.navigate(R.id.action_mainFragment_to_loginActivity)
                /*activity?.let {
                    *//*val intent = Intent(it, LoginActivity::class.java)
                    intent.putExtra("name", "admin")
                    intent.putExtra("psw", "123456")
                    it.startActivity(intent)*//*
                    toActivity<LoginActivity>(it) {
                        putExtra("name", "admin")
                        putExtra("psw", "123456")
                    }
                }*/
            }
        }
    }


}